src/amber/server/server.cr

```diff
@@ -67,9 +67,16 @@ module Amber
         exit
       end
 
-      logger.info "Server started in #{colorize(Amber.env, :yellow)}."
-      logger.info colorize("Startup Time #{Time.now - time}\n\n", :white)
-      server.listen(settings.port_reuse)
+      loop do
+        begin
+          logger.info "Server started in #{colorize(Amber.env, :yellow)}."
+          logger.info colorize("Startup Time #{Time.now - time}\n\n", :white)
+          server.listen(settings.port_reuse)
+        rescue e
```

Comment:
Can we rescue a specific error here, this is rescuing all errors. This means that if there is an error for some other reason the server will keep rebooting and failing. I will suggest to remove the loop.

Reply:
Rescuing a specific error makes sense. This is still a WIP as I haven't benchmarked changes yet.

Reply:
Best option would probably be to leave the loop but exit/raise the error inside or rescue unless it's the errors we want to retry.